pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -62,6 +62,14 @@ func newPodCache(c *Controller, informer coreinformers.PodInformer, queueEndpoin
 	return out
 }
 
+func isPodReady(pod *v1.Pod) bool {
+	ready := true
+	for _, cs := range pod.Status.ContainerStatuses {
+		ready = ready && cs.Ready
```

Comment:
Why do we need to do this, k8s endpoint controller does not sense each container status at all

Reply:
Are you saying that the pod object coming does not have the container readiness information? From my testing, this is not the case. I can see an unready pod is removed from the endpoint when I added this check.

Reply:
I am saying we donot need to care about this, the pod's phase is enough for us. 
Come back to this pr, i can not understand how this works

Reply:
Maybe I am having a limited understanding. But where does the pod condition get handled? ref: https://godoc.org/k8s.io/api/core/v1#PodConditionType

The reason I am adding this code is to only add the pods that are in `PodReady` condition into sidecar endpoints, which currently where the issues are.

Reply:
>The reason I am adding this code is to only add the pods that are in PodReady condition into sidecar endpoints, which currently where the issues are.

I think i understand what you mean here, it is same as what k8s endpoint_controller does https://github.com/kubernetes/kubernetes/blob/a744bddbf497ab718a40bee1c730e7e848e471ed/pkg/controller/endpoint/endpoints_controller.go#L612-L616

Reply:
But in the `pod.go`, every event regardless when the pod is ready or not, it will trigger a eds push though. In other words, a pod in `PodInitializing` state could also be sent as outbound endpoint.